pkg/actors/placement/client.go

```diff
@@ -44,43 +44,34 @@ type placementClient struct {
 	// streamConnectedCond is the condition variable for goroutines waiting for or announcing
 	// that the stream between runtime and placement is connected.
 	streamConnectedCond *sync.Cond
-
-	// ctx is the stream context
-	ctx context.Context
-	// cancel is the cancel func for context
-	cancel context.CancelFunc
 }
 
 // connectToServer initializes a new connection to the target server and if it succeeds replace the current
 // stream with the connected stream.
-func (c *placementClient) connectToServer(serverAddr string) error {
+func (c *placementClient) connectToServer(ctx context.Context, serverAddr string) error {
 	opts, err := c.getGrpcOpts()
 	if err != nil {
 		return err
 	}
 
-	conn, err := grpc.Dial(serverAddr, opts...)
+	conn, err := grpc.DialContext(ctx, serverAddr, opts...)
```

Comment:
If you use DialContext, it's important that you add the `WithBlock` option, since DialContext is non-blocking: https://pkg.go.dev/google.golang.org/grpc#DialContext